src/mpi/comm/comm_split_type.c

```diff
@@ -239,8 +299,17 @@ int MPIR_Comm_split_type_node_topo(MPIR_Comm * user_comm_ptr, int split_type, in
                        &errflag);
     if (mpi_errno)
         MPIR_ERR_POP(mpi_errno);
+    hintval_size = strlen(hintval);
+    mpi_errno =
+        MPIR_Allreduce(&hintval_size, &hintval_size_global, 1, MPI_INT, MPI_MAX, comm_ptr,
+                       &errflag);
```

Comment:
Are you simply comparing the size of the hints, and not the hints themselves?

Reply:
The initial check is for sizes, since buffer sizes for hints is unknown. Subsequently, I check the hint values.

Reply:
I see.  Thanks.  I worry that the checks are becoming too complex.  I'd recommend something similar to what you are doing for the `pci` case, but make it uniform for all cases (i.e., `pci`, `gpu`, ...).  So simply, check the `info_val_size` and `info_val` in all cases, and if they don't match ignore the info hint.

Reply:
Once you do that, you can get rid of the `hwloc` obj Allreduce.

Reply:
Makes sense, we can run 3 checks in total, two for `info_val_size` and `info_val` and one for hwloc_obj_type since we support aliases in processor object type hints (example: hwthread and pu). If we get rid of the aliases, the first two checks would suffice for all cases. I will refactor to perform 3 checks for now.

Reply:
I don't think we need to support the case where one process gives `hwthread` and another gives `pu`.  So two checks should suffice.